test/core/event_engine/forkable_test.cc

```diff
@@ -92,7 +100,44 @@ TEST_F(ForkableTest, BasicPthreadAtForkOperations) {
     }
   }
 }
-#endif  // GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK
+
+TEST_F(ForkableTest, NonPthreadManualForkOperations) {
+  // Manually simulates a fork event for non-pthread-enabled environments
+#ifdef GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK
+  return;
+#endif
+
+  class SomeForkable : public Forkable {
+   public:
+    void PrepareFork() override { prepare_called_ = true; }
+    void PostforkParent() override { parent_called_ = true; }
+    void PostforkChild() override { child_called_ = true; }
+
+    // bool prepare_called() { return prepare_called_; }
+    // bool parent_called() { return parent_called_; }
+    // bool child_called() { return child_called_; }
+
+    void AssertStates(bool prepare, bool parent, bool child) {
+      EXPECT_EQ(prepare_called_, prepare);
+      EXPECT_EQ(parent_called_, parent);
+      EXPECT_EQ(child_called_, child);
+    }
+
+   private:
+    bool prepare_called_ = false;
+    bool parent_called_ = false;
+    bool child_called_ = false;
+  };
+
+  SomeForkable forkable;
```

Comment:
does this test need the env var set?

Reply:
Oh good point. These tests now force-enable fork support, but exercise the Forkable with and without pthread support enabled. I think that's a fair compromise. By default, all other core & C++ tests exercise the forkable code when fork support is disable, so this is decent coverage.